main/main.cpp

```diff
@@ -269,6 +273,36 @@ void finalize_display() {
 	memdelete(display_server);
 }
 
+void initialize_navigation_mesh_generator() {
+	// Init chosen NavigationMeshGenerator
+	const String &server_name = GLOBAL_GET(NavigationMeshGeneratorManager::setting_property_name);
+	navigation_mesh_generator = NavigationMeshGeneratorManager::get_singleton()->new_server(server_name);
+
+	// Fall back to default if not found
+	if (!navigation_mesh_generator) {
+		// Navigation server not found, so use the default.
+		navigation_mesh_generator = NavigationMeshGeneratorManager::get_singleton()->new_default_server();
+	}
+
+	// Fall back to dummy if no default server has been registered.
+	if (!navigation_mesh_generator) {
+		ERR_PRINT("No NavigationMeshGenerator implementation has been registered! Falling back to a dummy implementation: navigation mesh baking features will be unavailable.");
+		navigation_mesh_generator = memnew(NavigationMeshGeneratorDummy);
+	}
+
+	if (navigation_mesh_generator) {
+		// need to register singleton earlier so modules / extensions / addons can use it on SCENE / SERVER init level
+		Engine::get_singleton()->add_singleton(Engine::Singleton("NavigationMeshGenerator", NavigationMeshGenerator::get_singleton()));
```

Comment:
The reason it ended up here is because those two mentioned function only run in normal Godot builds but would run to late or not run at all when Godot runs with tests enabled and I was in no mood to rework the entire test init chain.